ahmed-yara-sara/src/components/Search.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from "react";
+import { Spinner, Form, FormControl, Button } from "react-bootstrap";
+import "bootstrap/dist/css/bootstrap.min.css";
+export default function Search(props) {
+  const [isSubmitted, submitForm] = React.useState(false);
+  const [searchValue, setSearchValue] = React.useState("");
+  const TMDB_BASE_URL = "https://api.themoviedb.org/3";
+  const constructUrl = (path, query) => {
+    return `${TMDB_BASE_URL}/${path}?api_key=${atob(
+      "ZDJmYTdhZDFlMjZhZjA4NDdkMzQ5ZDdkYmQ1ZjkzZTU="
+    )}&query=${query}`;
+  };
+
+  const handleChange = (event) => {
+    setSearchValue(event.target.value);
+  };
+  const onSubmit = (e) => {
+    e.preventDefault();
+    submitForm(true);
+    fetchMoives();
+  };
+  React.useEffect(() => {
+    fetchMoives();
+    if (isSubmitted)
+      setTimeout(() => {
+        submitForm(false);
+      }, 3000);
+  }, [isSubmitted]);
+
+  function fetchMoives() {
+    let URL;
+    if (searchValue !== "") {
+      URL = constructUrl("search/movie", searchValue);
+    } else {
+      URL = constructUrl("movie/popular");
+    }
+    fetch(URL)
+      .then((movies) => movies.json())
+      .then((data) => {
+        props.getMoviesOnSearch(data.results);
+      });
+  }
+
+  return (
+    <Form inline onSubmit={onSubmit}>
+      <FormControl
+        type="text"
+        value={searchValue}
+        onChange={handleChange}
+        placeholder="Search"
+        className="mr-sm-2"
+      />
+      <Button variant="outline-info" type="submit">
+        {!isSubmitted && "Search"}
+        {isSubmitted && <Spinner size="sm" animation="border" variant="info" />}
```

Comment:
this could be better if you do this
```suggestion
        {!isSubmitted ? <Spinner size="sm" animation="border" variant="info" /> : "Search"}
```

but the `isSubmitted` should turn into `isLoading`